docs/input-logstash.asciidoc

```diff
@@ -236,64 +248,36 @@ A path to a JKS- or PKCS12-formatted keystore with which to identify this plugin
 * Value type is <<password,password>>
 * There is no default value for this setting.
 * Required when connection identity is configured with <<plugins-{type}s-{plugin}-ssl_certificate>>
-* {prohibit-ssl-disabled-effective}
 
 A password or passphrase of the <<plugins-{type}s-{plugin}-ssl_key>>.
 
-[id="plugins-{type}s-{plugin}-ssl_truststore_path"]
-===== `ssl_truststore_path`
+[id="plugins-{type}s-{plugin}-username"]
+===== `username`
 
-* Value type is <<path,path>>
+* Value type is <<string,string>>
 * There is no default value for this setting.
-* When present, <<plugins-{type}s-{plugin}-ssl_truststore_password>> is required.
-* {prohibit-ssl-disabled-effective}
-* {prohibit-ssl-verify-none}
-
-A path to JKS- or PKCS12-formatted keystore where trusted certificates are located.
-
-This setting can be used to _override_ the system trust store for verifying the SSL certificate presented by {es}.
 
-[id="plugins-{type}s-{plugin}-ssl_truststore_password"]
-===== `ssl_truststore_password`
-
-* Value type is <<password,password>>
-* There is no default value for this setting.
-* Required when connection trust is configured with <<plugins-{type}s-{plugin}-ssl_truststore_path>>
-* {prohibit-ssl-disabled-effective}
+Username for HTTP Basic Authentication.
```

Comment:
"HTTP Basic Authentication" is an internal implementation detail that is NOT meant to be propagated to the configuration boundary.

We _very much_ want to avoid putting implementation details into the configuration, because doing so prevents us from changing details in the future (**that in fact is the entire point of this plugin**).

We do not guarantee that communication between an upstream Logstash output and a downstream Logstash input will occur over HTTP. We only guarantee that an upstream Logstash output can send events to any supported and like-configured downstream Logstash input to which it has a TCP route. It is these very-specific promises that will allow us to iterate on the _internals_ of communication without breaking pipelines that begin using this output/input pair.

Reply:
Thanks for the clear explanation, agreed and updated.